governor.py

```diff
@@ -0,0 +1,408 @@
+import copy
+import inspect
+import logging
+import requests
+
+from collections import defaultdict, deque
+
+from config import _is_affirmative
+
+
+log = logging.getLogger('governor')
+
+
+class LimiterConfigError(Exception):
+    """
+    Error when parsing limiter
+    """
+    pass
+
+
+class Governor(object):
+    """
+    Abstract Governor class
+    """
+
+    # Defines what's an 'active' context
+    # i.e. number of iterations for which a context (not encountered anymore)
+    # is kept stored before being flushed.
+    _CONTEXTS_TTL = 3
+
+    # Collector payload's metric format
+    _METRIC_FORMAT = ['name', 'timestamp', 'value', 'attributes']
+
+    # Governor report endpoint
+    DATADOG_REPORT_GOVERNOR_URL = 'agent_governor/report_too_many_active_contexts'
```

Comment:
It would probably me more flexible to have a `agent_governor/report` endpoint and support different "events" from the governor